packages/edit-post/src/components/header/index.js

```diff
@@ -118,47 +106,19 @@ function Header( { setEntitiesSavedStatesCallback, initialPost } ) {
 				className="edit-post-header__toolbar"
 			>
 				<DocumentTools disableBlockTools={ isTextEditor } />
-				{ showTopToolbar && (
-					<>
-						<div
-							className={ classnames(
-								'selected-block-tools-wrapper',
-								{
-									'is-collapsed':
-										isBlockToolsCollapsed ||
-										! hasBlockSelection,
-								}
-							) }
-						>
-							<BlockToolbar hideDragHandle />
-						</div>
-						<Popover.Slot
-							ref={ blockToolbarRef }
-							name="block-toolbar"
-						/>
-						<Button
-							className="edit-post-header__block-tools-toggle"
-							icon={ isBlockToolsCollapsed ? next : previous }
-							onClick={ () => {
-								setIsBlockToolsCollapsed(
-									( collapsed ) => ! collapsed
-								);
-							} }
-							label={
-								isBlockToolsCollapsed
-									? __( 'Show block tools' )
-									: __( 'Hide block tools' )
-							}
-							size="compact"
-						/>
-					</>
+				{ hasTopToolbar && (
+					<ContextualToolbar
+						isCollapsed={ isBlockToolsCollapsed }
+						blockSelectionStart={ blockSelectionStart }
```

Comment:
Since `blockSelectionStart` isn't used it in this component, should we move it inside `ContextualToolbar`?